test/benchmarks/netperf/launch.go

```diff
@@ -0,0 +1,430 @@
+		}
+		name = fmt.Sprintf("netperf-w%d", i)
+		fmt.Println("Creating replication controller", name)
+		portSpec := []api.ContainerPort{}
+		if i > 1 {
+			// Worker W1 is a client-only pod - no ports are exposed
+			portSpec = append(portSpec, api.ContainerPort{ContainerPort: iperf3Port, Protocol: api.ProtocolTCP})
+		}
+		workerEnv := []api.EnvVar{
+			{Name: "worker", Value: name},
+			{Name: "kubeNode", Value: kubeNode},
+			{Name: "podname", Value: name},
+		}
+		_, err := c.ReplicationControllers(testNamespace).Create(&api.ReplicationController{
+			ObjectMeta: api.ObjectMeta{Name: name},
+			Spec: api.ReplicationControllerSpec{
+				Replicas: 1,
+				Selector: map[string]string{"app": name},
+				Template: &api.PodTemplateSpec{
+					ObjectMeta: api.ObjectMeta{
+						Labels: map[string]string{"app": name},
+					},
+					Spec: api.PodSpec{
+						NodeName: kubeNode,
+						Containers: []api.Container{
+							{
+								Name:            name,
+								Image:           netperfImage,
+								Ports:           portSpec,
+								Args:            []string{"--mode=worker"},
+								Env:             workerEnv,
+								ImagePullPolicy: "Always",
+							},
+						},
+						SecurityContext: &api.PodSecurityContext{HostNetwork: hostnetworking},
+					},
+				},
+			},
+		})
+		if err != nil {
+			fmt.Println("Error creating orchestrator replication controller", name, ":", err)
+			return false
+		}
+	}
+
+	return true
+}
+
+func getOrchestratorPodName(pods *api.PodList) string {
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.GetName(), "netperf-orch-") {
+			return pod.GetName()
+		}
+	}
+	return ""
+}
+
+// Retrieve the logs for the pod/container and check if csv data has been generated
+func getCsvResultsFromPod(c *client.Client, podName string) *string {
+	body, err := c.Pods(testNamespace).GetLogs(podName, &api.PodLogOptions{Timestamps: false}).DoRaw()
+	if err != nil {
+		fmt.Printf("Error (%s) reading logs from pod %s", err, podName)
+		return nil
+	}
+	logData := string(body)
+	index := strings.Index(logData, csvDataMarker)
+	endIndex := strings.Index(logData, csvEndDataMarker)
+	if index == -1 || endIndex == -1 {
+		return nil
+	}
+	csvData := string(body[index+len(csvDataMarker)+1 : endIndex])
+	return &csvData
+}
+
+// processCsvData : Process the CSV datafile and generate line and bar graphs
+func processCsvData(csvData *string) bool {
+	outputFilePrefix := fmt.Sprintf("%s-%s.", testNamespace, tag)
+	fmt.Printf("Test concluded - CSV raw data written to %s.csv\n", outputFilePrefix)
+	fd, err := os.OpenFile(fmt.Sprintf("%scsv", outputFilePrefix), os.O_RDWR|os.O_CREATE, 0666)
+	if err != nil {
+		fmt.Println("ERROR writing output CSV datafile", err)
+		return false
+	}
+	fd.WriteString(*csvData)
+	fd.Close()
+	return true
+}
+
+func executeTests(c *client.Client) bool {
+	for i := 0; i < iterations; i++ {
+		cleanup(c)
+		if !createServices(c) {
+			fmt.Println("Failed to create services - aborting test")
+			return false
```

Comment:
Its meant to be a standalone test that can be run against any Kubernetes cluster, I didn't want to entangle it with the e2e framework.